[GTK] webkit_settings_set_enable_2d_canvas_acceleration fails on s390x

mcatanzaro opened the pull request:
#### 4844fb011b186e6a8965174d77633769c49eac40
<pre>
[GTK] webkit_settings_set_enable_2d_canvas_acceleration fails on s390x
<a href="https://bugs.webkit.org/show_bug.cgi?id=279220">https://bugs.webkit.org/show_bug.cgi?id=279220</a>

Reviewed by Carlos Garcia Campos, Adrian Perez de Castro, and Fujii Hironori.

Cairo doesn&apos;t support hardware acceleration. (Too bad for Windows and
PlayStation and s390x!) So this setting only makes sense when Skia is
used, or on Apple platforms.

Also, the older setting no longer does anything, so it should default to
OFF.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
(webkit_settings_class_init):

Canonical link: <a href="https://commits.webkit.org/283863@main">https://commits.webkit.org/283863@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bdd04155cb31fee1861453a88688c04ad76d6407

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/67364 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/46743 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/19996 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/71412 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/18501 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/54541 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/18292 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/54021 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/12406 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/70431 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/42932 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/58272 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/34482 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/39604 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/15672 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/16859 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/60479 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/61572 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/16013 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/73111 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/66609 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/11323 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/15343 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/61456 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/11363 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/58337 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/61515 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/9275 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/2876 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/88378 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/10274 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/42548 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/15579 "Found 1047 jsc stress test failures: jsc-layout-tests.yaml/js/script-tests/function-apply-aliased.js.layout-no-cjit, microbenchmarks/wasm-cc-int-to-int.js.default-wasm, stress/json-stringify-inspector-check.js.no-llint, wasm.yaml/wasm/extended-const-spec-tests/elem.wast.js.default-wasm, wasm.yaml/wasm/extended-const-spec-tests/elem.wast.js.wasm-bbq ...") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/43625 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/44811 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/43366 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

fujii commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -1670,7 +1670,8 @@ CanvasUsesAcceleratedDrawing:
     WebKit:
-      default: true
+      "PLATFORM(COCOA) || USE(SKIA)": true
+      default: false
```
This setting is used only in `#if USE(CA) || USE(SKIA)`. How about adding `condition: #if USE(CA) || USE(SKIA)`?
https://github.com/WebKit/WebKit/blob/e4d5fc8e99945e2e1e6aed054d54fddb399c1a47/Source/WebCore/html/CanvasBase.cpp#L274-L275

mcatanzaro replied:
OK, good find.

mcatanzaro replied:
Updated.

fujii reviewed (approved):
LGTM. Might be better if GTK reviewers also approve.

carlosgcampos commented on Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
```diff
@@ -1307,7 +1307,11 @@ static void webkit_settings_class_init(WebKitSettingsClass* klass)
             FEATURE_DEFAULT(CanvasUsesAcceleratedDrawing),
+#else
+            FALSE,
+#endif
```
This is an existing issue, but this should always be FALSE, since this setting does nothing.

mcatanzaro replied:
Er, OK... but shouldn't we make it an alias of the newer setting?

(We probably should have just undeprecated the original setting rather than add the new one, but whatever.)

mcatanzaro replied:
Will change this to always FALSE before landing.

mcatanzaro committed:
[GTK] webkit_settings_set_enable_2d_canvas_acceleration fails on s390x
https://bugs.webkit.org/show_bug.cgi?id=279220

Reviewed by Carlos Garcia Campos, Adrian Perez de Castro, and Fujii Hironori.

Cairo doesn't support hardware acceleration. (Too bad for Windows and
PlayStation and s390x!) So this setting only makes sense when Skia is
used, or on Apple platforms.

Also, the older setting no longer does anything, so it should default to
OFF.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
(webkit_settings_class_init):

Canonical link: https://commits.webkit.org/283863@main
